src/transformState.js

```diff
@@ -5,7 +5,24 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
-}
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
 
+    switch (type) {
+      case 'addProperties':
+        Object.assign(state, extraData);
+        break;
+      case 'removeProperties':
+        for (const key of keysToRemove) {
+          delete state[key];
+        }
+        break;
+      default:
+        for (const key of Object.keys(state)) {
+          delete state[key];
+        }
+        break;
```

Comment:
```suggestion
      case 'clear':
        for (const key of Object.keys(state)) {
          delete state[key];
        }
        break;
      default:
        throw new Error('Unrecognised action.type');
```
While the `default` case should exist it doesn't mean that you should move the functionality of a specific case under this case.